datajoint/user_tables.py

```diff
@@ -186,3 +187,16 @@ def drop(self, force=False):
             super().drop()
         else:
             raise DataJointError('Cannot drop a Part directly.  Delete from master instead')
+
+    def alter(self, prompt=True, context=None):
+        """
+        Alter the table definition from self.definition
+        """
+        # map "master" keyword to master table in context
+        if context is None:
+            frame = inspect.currentframe().f_back
+            context = dict(frame.f_globals, **frame.f_locals)
+            del frame
+        if self.master:
+            context['master'] = self.master
+        super().alter(prompt, context)
```

Comment:
Nice find! I agree this is probably the best place to address this type of fix. However, you are right that there is a more elegant way to solve this problem.

The primary issue with your approach here is the additional overhead in inspecting the memory again (we've previously loaded this in `context`). Additionally, we add several more lines and it still doesn't satisfy further indexing into `master`.

Consider the following which should help address the above.
```suggestion
    def alter(self, prompt=True, context=None):
        super().alter(prompt=prompt, context=context or self.declaration_context)
```

Reply:
Thanks!